src/enr/crypto.ts

```diff
@@ -0,0 +1,55 @@
+import keccak = require("bcrypto/lib/keccak");
+import secp256k1 = require("bcrypto/lib/secp256k1");
+
+import {
+  NodeId,
+  PrivateKey,
+  PublicKey,
+} from "./types";
+
+export function hash(input: Buffer): Buffer {
+  return keccak.digest(input);
+}
+
+export function createPrivateKey(): PrivateKey {
+  return secp256k1.privateKeyGenerate();
+}
+
+export function publicKey(privKey: PrivateKey): PublicKey {
+  return secp256k1.publicKeyCreate(privKey);
+}
+
+export function sign(privKey: PrivateKey, msg: Buffer): Buffer {
+  return secp256k1.sign(
+    hash(msg),
+    privKey,
+  );
+}
+
+export function verify(pubKey: PublicKey, msg: Buffer, sig: Buffer): boolean {
+  return secp256k1.verify(hash(msg), sig, pubKey);
+}
+
+export function nodeId(pubKey: PublicKey): NodeId {
+  return hash(secp256k1.publicKeyConvert(pubKey, false));
+}
+
+export class ENRKeyPair {
+  public readonly nodeId: NodeId;
+  public readonly privateKey: PrivateKey;
+  public readonly publicKey: PublicKey;
+
+  public constructor() {
+    this.privateKey = createPrivateKey();
+    this.publicKey = publicKey(this.privateKey);
+    this.nodeId = nodeId(this.publicKey);
+  }
```

Comment:
Feels like you could use constructor which accepts private key (just a guess)

Reply:
Ah, yeah, will add. Currently not using this class, but it could be useful.